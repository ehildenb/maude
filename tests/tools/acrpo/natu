#!/bin/sh

MAUDE_LIB=$srcdir/../../../src/Main
export MAUDE_LIB

../../../src/Main/maude \
  < natu.maude -no-banner -no-advise \
  > natu.out 2>&1

diff natu.expected natu.out > /dev/null 2>&1
