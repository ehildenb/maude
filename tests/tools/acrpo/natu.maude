set show timing off .

fmod NATU is sort Natu .
  op 0     : -> Natu [ctor] .
  op s     : Natu -> Natu [ctor] .
  op _+_   : Natu Natu -> Natu [comm] .
  op _*_   : Natu Natu -> Natu .
  op _^_   : Natu Natu -> Natu .

  eq X:Natu + 0 = X:Natu .
  eq X:Natu + s(Y:Natu) = s(X:Natu + Y:Natu) .
  eq X:Natu * 0 = (0).Natu .
  eq X:Natu * s(Y:Natu) = (X:Natu * Y:Natu) + X:Natu .
  eq X:Natu ^ 0 = (s(0)).Natu .
  eq X:Natu ^ s(Y:Natu) = X:Natu * (X:Natu ^ Y:Natu) .
endfm

set include BOOL off .

--- NATU Signature
fmod TEST is
  sort Natu .
  sort U . 
  subsorts Natu < U .
  op 0 : -> Natu [ctor metadata "1"] .
  op s : Natu -> Natu [ctor metadata "2"] .
  op _+_ : Natu Natu -> Natu [comm metadata "3"] .
  op _*_ : Natu Natu -> Natu [metadata "4"] .
  op _^_ : Natu Natu -> Natu [metadata "5"] .
endfm

load ../../../contrib/tools/acrpo/acrpo.maude

red X:Natu + 0 >AC X:Natu .
red X:Natu + s(Y:Natu) >AC s(X:Natu + Y:Natu) .
red X:Natu * 0 >AC (0).Natu .
red X:Natu * s(Y:Natu) >AC (X:Natu * Y:Natu) + X:Natu .
red X:Natu ^ 0 >AC (s(0)).Natu .
red X:Natu ^ s(Y:Natu) >AC X:Natu * (X:Natu ^ Y:Natu) .
