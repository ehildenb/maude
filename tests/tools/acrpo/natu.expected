==========================================
reduce in ACRPO-EXT : (0 + X:Natu) >AC X:Natu .
rewrites: 3123
result @#Bool#@: @#true#@
==========================================
reduce in ACRPO-EXT : (X:Natu + s(Y:Natu)) >AC s(X:Natu + Y:Natu) .
rewrites: 332
result @#Bool#@: @#true#@
==========================================
reduce in ACRPO-EXT : (X:Natu * 0) >AC 0 .
rewrites: 54
result @#Bool#@: @#true#@
==========================================
reduce in ACRPO-EXT : (X:Natu * s(Y:Natu)) >AC (X:Natu + (X:Natu * Y:Natu)) .
rewrites: 284
result @#Bool#@: @#true#@
==========================================
reduce in ACRPO-EXT : (X:Natu ^ 0) >AC s(0) .
rewrites: 91
result @#Bool#@: @#true#@
==========================================
reduce in ACRPO-EXT : (X:Natu ^ s(Y:Natu)) >AC (X:Natu * (X:Natu ^ Y:Natu)) .
rewrites: 281
result @#Bool#@: @#true#@
Bye.
